### DISCUSSION TITLE
datetime mapping consistency

### DISCUSSION THREAD

**Discussion by Triquetra** [emmett-framework/emmett#353] [General]
The [Models Documentation](https://emmett.sh/docs/2.2.x/orm/models) indicates that datetime fields are mapped to datetime.datetime python objects. But the [Validations Documentation](https://emmett.sh/docs/2.2.x/validations#type-of-input) indicates that "the datetime validator returns a pendulum Datetime object." This inconsistency can cause confusion. I think it would be useful to consistently use either the standard library datetime or pendulum.datetime objects. Personally, I would prefer pendulum.datetime objects.

**Comment by gi0baro** [comment]
In general there shouldn't be issues at all, since `pendulum.Datetime` class is a subclass of `datetime.datetime`, and any validation performed by Emmett checks the object is an instance of `datetime.datetime`, thus any subclass is still valid.

In general datetime objects built by emmett, like the `now` function or the params parsed by router, are pendulum objects. The only place in which datetime objects are not automatically converted is the ORM, so when you select `Row` objects, they contains `datetime.datetime` objects, due to a couple of reasons:

- avoid slowing down ORM's parsing code, to inject the new objects
- some database drivers generate `datetime` objects through C code or extensions, and thus the only way would be re-parsing in Python the whole recordset.

**Comment by Triquetra** [comment]
Thank you for the explanation. That helps explain the design decision. I hadn't used pendulum before using emmett, but I like it. I had tried to use a Row.datetime object as though it was a pendulum datetime, but obviously, that didn't work... I just had to recast it.
